File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int ans =0;
        for(int i =0;i<nums.length && i<=ans;i++){
            ans = Math.max(ans,Math.min(i+nums[i],nums.length-1));
        }
        return ans == nums.length-1;
    }
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxJump = 0
        for i in range(len(nums)):
            j = i + nums[i]
            if maxJump < i:
                return False
            maxJump = max(maxJump,j)
        return True
